app/src/main/java/org/android/go/sopt/SignupActivity.kt

```diff
@@ -0,0 +1,44 @@
+package org.android.go.sopt
+
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.google.android.material.snackbar.Snackbar
+import org.android.go.sopt.databinding.ActivitySignupBinding
+
+class SignupActivity : AppCompatActivity() {
+    lateinit  var binding : ActivitySignupBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignupBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        clickSignUp_end()
+    }
+
+    private fun clickSignUp_end(){
+
+        binding.btnSignupEnd.setOnClickListener {
+            if(binding.etName.length() > 0 &&binding.etID.length()in 6..10 && binding.etPW.length()in 8..12 && binding.etHobby.length() > 0 ){
```

Comment:
문자열이 공백이 아니길 바라는거라면, 아래와 같이 작성할 수 있습니다! 참고해보세요 ~ 
https://thecommelier.tistory.com/16
```suggestion
            if(binding.etName.isNotBlank() &&binding.etID.length()in 6..10 && binding.etPW.length()in 8..12 && binding.etHobby.isNotBlank() ){
```